Libplanet/Store/TrieStateStore.cs

```diff
@@ -30,76 +24,30 @@ public class TrieStateStore : IStateStore
         /// </summary>
         /// <param name="stateKeyValueStore">The storage to store states. It used by
         /// <see cref="MerkleTrie"/> in internal.</param>
-        /// <param name="stateHashKeyValueStore">The storage to store state hash corresponding to
-        /// block hash.</param>
         /// <param name="secure">Whether to use <see cref="MerkleTrie"/> in secure mode.
         /// <seealso cref="MerkleTrie(IKeyValueStore, INode?, bool)"/>.</param>
         public TrieStateStore(
             IKeyValueStore stateKeyValueStore,
-            IKeyValueStore stateHashKeyValueStore,
             bool secure = false)
         {
             _stateKeyValueStore = stateKeyValueStore;
-            _stateHashKeyValueStore = stateHashKeyValueStore;
             _secure = secure;
             _logger = Log.ForContext<TrieStateStore>();
         }
 
-        /// <inheritdoc/>
-        public void SetStates<T>(
-            Block<T> block,
-            IImmutableDictionary<string, IValue> states)
-            where T : IAction, new()
-        {
-            _stateHashKeyValueStore.Set(
-                block.Hash.ToByteArray(), EvalState(block, states).ToByteArray());
-        }
-
-        /// <inheritdoc/>
-        public IValue? GetState(string stateKey, BlockHash? blockHash = null)
-        {
-            if (blockHash is null)
-            {
-                throw new ArgumentNullException(nameof(blockHash));
-            }
-
-            var stateHash = _stateHashKeyValueStore.Get(blockHash?.ToByteArray()!);
-            var stateTrie = new MerkleTrie(
-                _stateKeyValueStore, new HashNode(new HashDigest<SHA256>(stateHash)), _secure);
-            var key = Encoding.UTF8.GetBytes(stateKey);
-            return stateTrie.TryGet(key, out IValue? value) ? value : null;
-        }
-
-        /// <inheritdoc/>
-        public bool ContainsBlockStates(BlockHash blockHash)
-        {
-            return _stateHashKeyValueStore.Exists(blockHash.ToByteArray());
-        }
-
-        /// <inheritdoc/>
-        public void ForkStates<T>(Guid sourceChainId, Guid destinationChainId, Block<T> branchpoint)
-            where T : IAction, new()
-        {
-            // Do nothing.
-        }
-
-        public void PruneStates(IImmutableSet<BlockHash> excludeBlockHashes)
+        /// <inheritdoc cref="IStateStore.PruneStates(IImmutableSet{HashDigest{SHA256}})"/>
+        public void PruneStates(IImmutableSet<HashDigest<SHA256>> stateRootHashes)
         {
             var stopwatch = new Stopwatch();
             _logger.Verbose($"Started {nameof(PruneStates)}()");
             var excludeNodes = new HashSet<HashDigest<SHA256>>();
-            foreach (var blockHash in excludeBlockHashes)
+            foreach (HashDigest<SHA256> stateRootHash in stateRootHashes)
```

Comment:
The argument's name was renamed but the logic seems not changed. 🤔 Is it okay? Originally the name was `excludeBlockHashes` and `PruneStates` removed all states except others included in the given `excludeBlockHashes`.

According to the `IStateStore.PruneStates`' document, it should prune the all states included in the given `stateRootHashes` if I understood correctly, without effect to other state roots 🤔

Reply:
It's apparently my mistake.  The word *exclude* here seemed ambiguous anyway, because it can mean both opposite ways:

1. to *exclude* (= delete) them from the state store,
2. or to *exclude* them from removal (= remain them).

During changing the code, I misunderstood its intention — I thought it meant the latter.  However, it's turned out that it meant the former, right?

As it's more misleading with the current name (`stateRootHashes`) than before, I'm going to rename it anyway, but because its original name was also misleading, I suggest a new name: `survivalStateRootHashes`.  Although it's quite long, seems more descriptive to me.  Would this name satisfy you too?

Reply:
I thought a few minutes but I don't have better naming 😞 

It seems okay 👌🏻 👍🏻

Reply:
Applied the new name (`survivalStateRootHashes`): d65fe700f93f09d47f5d06497e459335ad8b8beb.  Feel free to suggest any better name even later.